as/byte-buffer.ts

```diff
@@ -0,0 +1,365 @@
+import { FILE_IDENTIFIER_LENGTH, SIZEOF_INT } from './constants';
+import { Long } from './long';
+import { int32, isLittleEndian, float32, float64 } from './utils';
+import { Offset, Table, IGeneratedObject } from './types';
+
+export class ByteBuffer {
+  private position_: i32 = 0;
+
+  /**
+   * Create a new ByteBuffer with a given array of bytes (`Uint8Array`)
+   */
+  constructor(private bytes_: Uint8Array) {}
+
+  /**
+   * Create and allocate a new ByteBuffer with a given size.
+   */
+  static allocate(byte_size: i32): ByteBuffer {
+    return new ByteBuffer(new Uint8Array(byte_size));
+  }
+
+  clear(): void {
+    this.position_ = 0;
+  }
+
+  /**
+   * Get the underlying `Uint8Array`.
+   */
+  bytes(): Uint8Array {
+    return this.bytes_;
+  }
+
+  /**
+   * Get the buffer's position.
+   */
+  position(): i32 {
+    return this.position_;
+  }
+
+  /**
+   * Set the buffer's position.
+   */
+  setPosition(position: i32): void {
+    this.position_ = position;
+  }
+
+  /**
+   * Get the buffer's capacity.
+   */
+  capacity(): i32 {
+    return this.bytes_.length;
+  }
+
+  readInt8(offset: i32): i32 {
+    return (i32(this.readUint8(offset)) << 24) >> 24;
```

Comment:
Manual sign extension is unnecessary when you cast to `u8 -> i32` via `as i32`
```suggestion
    return this.readUint8(offset) as i32;
```